source/agora/utils/TxBuilder.d

```diff
@@ -472,33 +584,58 @@ public struct TxBuilder
         return this;
     }
 
-    /***************************************************************************
-
-        Deduct all the remaining amount, so there will be no refund transaction
-
-        Useful when trying to test fee calculation
-
-        Returns:
-            Reference to `this` for easy chaining
-
-    ***************************************************************************/
-
-    public ref typeof(this) deductRemaining () pure nothrow @safe @nogc
-        return
-    {
-        this.leftover.value = Amount(0);
-
-        return this;
-    }
+    /// Any refund less than this amount will not create a refund output but be
+    /// left to be included as fees.
+    private const MinRefundAmount = Amount(500_000);
 
     /// Refund output for the transaction
     private Output leftover;
 
+    // fee per byte rate to be paid for the tx
+    private Amount fee_rate = Amount(700);
+
     /// Stores the inputs to consume until `sign` is called
     private OutputRef[] inputs;
 
     /// Transactions to be built and returned
     private Transaction data;
+
+    /// Unlocker function to sign the inputs
+    private Unlock function (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow unlocker;
```

Comment:
Maybe add:
```D
/// Type of function called by `TxBuilder.sign` to generate the unlock script
public alias Unlocker = Unlock function (in Transaction, in OutputRef) @safe nothrow;
```

So that we don't have to repeat the same type everywhere (I think we're a bit past the rule of 3 :) ).